### ISSUE TITLE
More extensive documentation.

### ISSUE THREAD

**Issue by cmartinotti** [jpmorganchase/ipyregulartable#68]
This seems an amazing product, but nowhere is clearly explained exactly how you are supposed to update the data in the table, how to access the underlying data structures, and much more.  As an example, I'm trying to make it so that if I click on a bool type column of the table then the data structure is change to a new data structure with the bool inverted.  

I guess the solution is to pass through  get_state to obtain the data, then setData to swap the underlying dataframe, but there is no doc string on any of these functions. It would be real nice to have some description of what these function do.

**Comment by timkpaine** [comment]
Best bet is to start with the examples:
https://github.com/jpmorganchase/ipyregulartable/tree/main/docs/examples

There is a relatively simple example of a data models here:
https://github.com/jpmorganchase/ipyregulartable/blob/main/ipyregulartable/datamodel/demo.py

The best way to force an update on the frontend is by calling `draw` on the widget (should happen automatically if the data model changes):
https://github.com/jpmorganchase/ipyregulartable/blob/main/ipyregulartable/widget.py#L110

I haven't written much documentation as the codebase is still pretty alpha, I have a major overhaul in progress in https://github.com/jpmorganchase/ipyregulartable/pull/59 to add some nice defaults like striping, cursor navigation of cells, column/row/both selection, sortable headers, etc. Once that overhaul is done I think the codebase will be in a much better and more stable place and then can have some more extensive docs written.

**Comment by cmartinotti** [comment]
Thanks for the reply and amazing work! Is ipyregulartable compatible with ipywidgets boxes or will it be in the future?  I tried to capture it with a box and with an Output widget but had no luck.
